### ISSUE TITLE
Option to disable SQL Server health check delay

### ISSUE THREAD

**Issue by OskarKlintrot** [Azure/azure-service-bus-emulator-installer#50]
**Is your feature request related to a problem? Please describe.**
When starting the container, it will pause and wait 15 sec before checking in on SQL Server. I'm running SQL Server locally so there is no need to wait for the Azure SQL Edge container to start (since I don't even use it).

**Describe the solution you'd like**
Use for example an env var to allow us to disable the wait:
```
"DISABLE_WAIT_BEFORE_SQL_SERVER_HEALTH_CHECK=true"
```

**Describe alternatives you've considered**
Making coffee.

**Additional context**
From the log:
```
2024-12-06 09:54:32 info: EmulatorHealthCheckHelper[0]
2024-12-06 09:54:32       Waiting for 15 seconds before starting the health check for SQL
```

**Comment by neffsvg** [comment]
This is a real problem when writing integration tests - its way too long.

There is another issue which is a duplicate of this one: https://github.com/Azure/azure-service-bus-emulator-installer/issues/15

If you want to keep the default 15 Seconds, one could add
`"DELAY_STARTUP_FOR_HEALTH_CHECKS_IN_SECONDS=0"`
and if set to 0 just skip the timeout, if var is not set keep the default 15.

If a breaking change is an option here, one could introduce an optional var:
`"DELAY_STARTUP_IN_SECONDS=15"`


When writing integration tests, one would want to start the sql, then the sb container before each test. In those scenarios any delay / blocking timeout is a waste of time/resources, one would want to ensure containers are started by code.

We are using golang and just create both containers after each other, so that timeout is probably not at all necessary, or could be added to code.

**Comment by todthomson** [comment]
👍 to this suggestion. It provides an unnecessary delay when there are plenty of other ways to make sure that SQL Server is available before starting up the emulator.

**Comment by reinkrul** [comment]
We also run into this.

**Comment by mdelapenya** [comment]
I commented in the original issue that the ideas proposed here could work for testcontainers: https://github.com/Azure/azure-service-bus-emulator-installer/issues/15#issuecomment-2703382717

**Comment by krishankumar95** [comment]
SQL_WAIT_INTERVAL docker environment variable has been introduced in the latest release to allow for reducing/disabling initial wait for SQL Health Check. The value is in seconds. 

More details about latest release can be found at : [Changelog](https://learn.microsoft.com/en-us/azure/service-bus-messaging/service-bus-emulator-whats-new#latest-version-112)

**Comment by OskarKlintrot** [comment]
@danny12321 It's already available. I think 4 months ago is when the first image was released.

**Comment by krishankumar95** [comment]
@danny12321 ; Kindly refer: [MCR AZ ASB Emulator Tags](https://mcr.microsoft.com/en-us/artifact/mar/azure-messaging/servicebus-emulator/tags)
 
Do take a pull of the installer repo as well; it includes a pull policy update in Docker compose file.

**Comment by danny12321** [comment]
@OskarKlintrot & @krishankumar95 thanks for the replies. I didn't find the Microsoft Artifact Registry by googling, so the link helped a lot. Thanks!

**Comment by asos-iqan** [comment]
can use healthcheck and SQL_WAIT_INTERVAL to reduce overall delay.
```yaml
servicebus-db:
    image: mcr.microsoft.com/azure-sql-edge:latest
    environment:
      ACCEPT_EULA: Y
      MSSQL_SA_PASSWORD: ${SQL_PASS}
    healthcheck:
      test: /opt/mssql-tools/bin/sqlcmd -C -U sa -Q "SELECT 1"
      interval: 5s
      timeout: 3s
      retries: 5

  servicebus:
    image: mcr.microsoft.com/azure-messaging/servicebus-emulator:latest
    pull_policy: always
    volumes:
      - ${PATH_TO_CONFIG}:/ServiceBus_Emulator/ConfigFiles/Config.json
    ports:
      - "5672:5672"
      - "5300:5300"
    environment:
      SQL_SERVER: servicebus-db
      MSSQL_SA_PASSWORD: ${SQL_PASS}
      ACCEPT_EULA: Y
      SQL_WAIT_INTERVAL: 1
    depends_on:
      servicebus-db:
        condition: service_healthy     # waits for db healthcheck to be successful
```
